src/main/java/org/darisadesigns/polyglotlina/DictCore.java

```diff
@@ -456,9 +362,10 @@ public void readFile(String _fileName, byte[] overrideXML) throws IOException, I
         }
         
         // do not run in headless environments...
-        if (rootWindow != null) {
-            refreshMainMenu();
-        }
+        // TODO: replace with signal?
+//        if (polyGlot.getRootWindow() != null) {
```

Comment:
Again, this is really important. Gonna have to think over the best way to address this once I've merged to this branch and can play around with it locally.

Reply:
Fixed.
Implemented a listener that makes the same call.